src/tree/updater_quantile_hist.h

```diff
@@ -186,6 +186,13 @@ class QuantileHistMaker: public TreeUpdater {
       unsigned timestamp;
       ExpandEntry(int nid, int sibling_nid, int depth, bst_float loss_chg, unsigned tstmp):
         nid(nid), sibling_nid(sibling_nid), depth(depth), loss_chg(loss_chg), timestamp(tstmp) {}
+
+      bool IsValid(TrainParam const& param, int32_t num_leaves) const {
```

Comment:
This looks duplicated with GPU code, can we share this expand entry struct?

Reply:
Actually I pushed this change to avoid reading the very long boolean expression.